### PR TITLE
#11 JWT auth implemented

### PR DESCRIPTION
**Pull request by alex-a-renoire**

### CODE REVIEW

File: service/http/service.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+		var publickey tokenResp
+		if err = json.NewDecoder(resp.Body).Decode(&publickey); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal public key: %w", err)
+		}
+
+		block, _ := pem.Decode(publickey.PbKey)
+		pubInterface, err := x509.ParsePKIXPublicKey(block.Bytes)
+
+		publicKey, ok := pubInterface.(*rsa.PublicKey)
```
Comment (DrIhor) [inline service/http/service.go]: Update names. You have publickey using type(tokenResp) and type(rsa.PublicKey)

File: service/http/service.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to parse token: %w", err)
+	}
+
+	if !token.Valid {
+		return fmt.Errorf("token is not valid: %w", err)
+	}
+
```
Comment (DrIhor) [inline service/http/service.go:80]: Can add token expire validation.

File: service/http/service.go (hunk 1 above)
Comment (DrIhor) [reply service/http/service.go]: Diff is K letter in word but it very confusing

**Review by DrIhor — APPROVED** [review]

File: cmd/http/main.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ func getCfg() Config {
 
 	jwtAddr := os.Getenv("JWT_ADDRESS")
 	if jwtAddr == "" {
-		jwtAddr = "127.0.0.1:8081"
+		jwtAddr = "127.0.0.1:8081/get-cert"
```
Comment (romanilchyshyn) [inline cmd/http/main.go]: Lets differentiate jwt address and specific endpoint. You can generate specific endpoint from addr + path at any place needed.

**Review by romanilchyshyn — APPROVED** [review]

**Pull request merged**